statsmodels/survey/survey_model.py

```diff
@@ -0,0 +1,146 @@
+            e.append(np.dot(w[ind] * d_hat[:,ind], np.dot((y - cond_mean)[ind], cond_inv)))
+        e = np.asarray(e)
+        e = self._centering(e, 'stratum')
+        nh = self.design.clust_per_strat[self.design.strat_for_clust].astype(np.float64)
+
+        mh = np.sqrt(nh / (nh - 1))
+        fh = np.sqrt(1 - self.design.fpc)
+        e = fh[:, None] * mh[:, None] * e
+        g_hat = (len(y) - 1) / (len(y) - X.shape[1]) * np.dot(e.T, e)
+
+        if technique=="newton":
+            # TODO: Figure out if hessian should be used when
+            # 'jack-sandwich' or 'boot-sandwich' is requested
+            model.fit(**self.fit_args)
+            factor = model.hessian_factor(self.params)
+            q_hat = -np.dot(X.T * factor, X)
+        else:
+            cond_inv = np.linalg.inv(np.diag(cond_mean) - np.dot(cond_mean, cond_mean.T))
+            q_hat = np.dot(w * d_hat, np.dot(cond_inv, d_hat.T))
+        q_hat_inv = np.linalg.inv(q_hat)
+        vcov = np.dot(q_hat_inv, np.dot(g_hat, q_hat_inv))
+        return vcov
+
+    def _get_jackknife_vcov(self, X, y):
+        replicate_params = []
+        for c in range(self.design.n_clust):
+            w = self.design.get_rep_weights(cov_method='jack', c=c)
+            self.init_args["weights"] = w
+            replicate_params.append(self._get_params(y, X))
+        replicate_params = np.asarray(replicate_params)
+        self.replicate_params = self._centering(replicate_params)
+        try:
+            nh = self.design.clust_per_strat[self.design.strat_for_clust].astype(np.float64)
+            mh = np.sqrt((nh - 1) / nh)
+            self.replicate_params = mh[:, None] * self.replicate_params
+        except AttributeError:
+            nh = self.design.rep_weights.shape[1]
+            mh = np.sqrt((nh - 1) / nh)
+            self.replicate_params *= mh
+        vcov = np.dot(self.replicate_params.T, self.replicate_params)
+        return vcov
+
+    def fit(self, y, X, cov_method='jack', center_by='est', technique=None):
+        self.center_by = center_by
+        self.init_args["weights"] = self.design.weights
+        self.params = self._get_params(y, X)
+
+
+        # for now, just working with jackknife to see if it works
+        if cov_method == 'jack':
+            self.vcov = self._get_jackknife_vcov(X, y)
+        elif cov_method == 'linearized_sas':
+            self.vcov = self._sas_linearization_vcov(X, y, technique)
+        elif cov_method == 'linearized_stata':
+            self.vcov = self._stata_linearization_vcov(X, y)
+        else:
+            return ValueError('cov_method %s not supported' %cov_method)
+        if self.vcov.ndim == 2:
+            self.stderr = np.sqrt(np.diag(self.vcov))
+        else:
+            self.stderr = np.sqrt(self.vcov)
+
+    def _get_params(self, y, X):
+        model = self.model(y, X, **self.init_args)
+        self.result = model.fit(**self.fit_args)
+        return self.result.params
```

Comment:
Yes, this is only because I need other attributes of self.result later. Will be sure to do this sparingly